cfg/builder/builder_walk.cc

```diff
@@ -257,8 +257,6 @@ BasicBlock *CFGBuilder::walk(CFGContext cctx, ast::ExpressionPtr &what, BasicBlo
                 if (thenEnd != cctx.inWhat.deadBlock() || elseEnd != cctx.inWhat.deadBlock()) {
                     if (thenEnd == cctx.inWhat.deadBlock()) {
                         ret = elseEnd;
-                    } else if (thenEnd == cctx.inWhat.deadBlock()) {
-                        ret = thenEnd;
```

Comment:
I think it is more correct to change this to test `elseEnd` -- both for symmetry and for not introducing superfluous blocks in the case that `elseEnd` is dead.  That change passes locally for me as well, which is kind of unfortunate.